### ISSUE TITLE
Allow mysql-router-k8s to be exposed outside of Kubernetes cluster

### ISSUE THREAD

**Issue by phvalguima** [canonical/mysql-router-k8s-operator#137]
Mysql router should provide means to provide access outside of the kubernetes cluster.

That can be achieved with a load balancer, for example:
```
$ cat <<EOF | kubectl apply -f -
apiVersion: v1
kind: Service
metadata:
  name: mysql-router
  namespace: mysql
spec:
  type: LoadBalancer
  ports:
  - name: mysql-rw
    targetPort: 6446
    port: 6446
    protocol: TCP
  - name: mysql-ro
    port: 6447
    protocol: TCP
    targetPort: 6447
  selector:
    app.kubernetes.io/name: mysql-router
EOF
```

**Comment by phvalguima** [comment]
Hi @carlcsaposs-canonical, unfortunately, no. Data integrator sets up the credentials, but we need to actually expose the mysql-router-k8s service.

**Comment by taurus-forever** [comment]
> Does juju expose help with that?

AFAIK, no. `juju expose` [is about firewall](https://discourse.charmhub.io/t/juju-k8s-charm-expose/1250/3?u=taurus).
This should do the job `juju deploy mysql-router-k8s --config kubernetes-service-type=LoadBalancer`
but "it doesn't work (c) @phvalguima"

**Comment by phvalguima** [comment]
Hi @taurus-forever, I also thought we could use that config, however, Juju team informed me that is only available for an older format of container apps called ```podspec``` in Juju. I also tried myself, and it did not work on k8s.
Here is the full chat: https://chat.charmhub.io/charmhub/pl/o6kqtisokjdd3n9wenhew4icfc

So, the only way is to expose is by patching the k8s service itself.

**Comment by phvalguima** [comment]
yes, if we have the lb option, we can close it @carlcsaposs-canonical @taurus-forever
